packages/preview-middleware-client/src/flp/init.ts

```diff
@@ -192,21 +193,39 @@ export async function init({ appUrls, flex }: { appUrls?: string | null; flex?:
         sap.ushell.Container.attachRendererCreatedEvent(async function () {
             const lifecycleService = await sap.ushell.Container.getServiceAsync<AppLifeCycle>('AppLifeCycle');
             lifecycleService.attachAppLoaded((event) => {
+                const version = sap.ui.version;
+                const minor = Number(version.split(/[.-]/)[1]);
                 const view = event.getParameter('componentInstance');
-                const libs = ['sap/ui/rta/api/startAdaptation'];
                 const flexSettings = JSON.parse(flex);
+                const pluginScript = flexSettings.pluginScript ?? '';
+                const initRtaScript = 'open/ux/preview/client/flp/initRta';
+                const libs = ['sap/ui/rta/api/startAdaptation'];
+
                 if (flexSettings.pluginScript) {
-                    libs.push(flexSettings.pluginScript);
+                    libs.push(pluginScript);
                     delete flexSettings.pluginScript;
                 }
-                sap.ui.require(libs, function (startAdaptation: StartAdaptation, pluginScript?: RTAPlugin) {
-                    const options = {
-                        rootControl: view,
-                        validateAppVersion: false,
-                        flexSettings
-                    };
-                    startAdaptation(options, pluginScript);
-                });
+
+                const options: RTAOptions = {
+                    rootControl: view,
+                    validateAppVersion: false,
+                    flexSettings
+                };
+
+                if (minor > 71) {
+                    sap.ui.require(libs, function (startAdaptation: StartAdaptation, pluginScript?: RTAPlugin) {
+                        startAdaptation(options, pluginScript);
+                    });
+                } else {
+                    // For ui5 version 1.71.x startAdaptation will fail (because 'sap/ui/rta/api/startAdaptation' does not exists -> the plugin script is never executed)
+                    // thus we need to initialize and start rta ourselves.
+                    sap.ui.require(
+                        [initRtaScript, pluginScript],
+                        async function (initRta: InitRtaScript, pluginScript: RTAPlugin) {
+                            await initRta(options, pluginScript);
+                        }
+                    );
+                }
```

Comment:
Could you please refactor the code to 
```
sap.ui.require(libs, function (startAdaptation: StartAdaptation | InitRtaScript, pluginScript?: RTAPlugin) {
                        startAdaptation(options, pluginScript);
                    });
```

Hence according to the UI5 version we set the `libs`, instead of having different `sap.ui.require` im code.

Reply:
Changed in: [3f1f9d5](https://github.com/SAP/open-ux-tools/pull/1597/commits/3f1f9d5314445754b1a015564e7a6480404ebc90)

Reply:
thx